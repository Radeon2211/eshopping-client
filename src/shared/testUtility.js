// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const historyPageNum = (pageNumber) => ({
  listen: jest.fn(),
  location: { search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push: jest.fn(),
});

export const propsPagination = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  maxQuantityPerPage: 2,
});
