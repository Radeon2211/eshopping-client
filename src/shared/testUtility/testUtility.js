// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';
import { v4 as uuidv4 } from 'uuid';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../../store/reducers/rootReducer';
import { productConditions } from '../constants';

export const defaultDeliveryAddress = {
  firstName: 'firstName',
  lastName: 'lastName',
  street: 'street',
  zipCode: 'zipCode',
  city: 'city',
  country: 'country',
  phone: 'phone',
};

export const defaultUserProfile = {
  ...defaultDeliveryAddress,
  _id: 'userId',
  username: 'username',
  email: 'email@domain.com',
  status: 'active',
  contacts: {
    email: true,
    phone: true,
  },
  createdAt: '2021-02-10T19:10:38.872Z',
};

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const createHistoryPageNumber = (pageNumber = 1, push = jest.fn()) => ({
  listen: jest.fn(),
  location: { pathname: '/products', search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push,
});

export const createPaginationProps = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  quantityPerPage: 2,
});

export const createProductItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
  quantitySold = 0,
  buyerQuantity = 0,
  description = '',
  condition = productConditions.NEW,
) => ({
  _id: productId || uuidv4(),
  name,
  price,
  quantity,
  quantitySold,
  buyerQuantity,
  description,
  condition,
  photo,
  seller: {
    username: sellerUsername,
  },
  __v: 0,
  createdAt: '2021-02-10T19:10:38.872Z',
  updatedAt: '2021-02-10T19:10:38.872Z',
});

export const createCartItem = (
  sellerUsername = 'username',
  quantity = 1,
  productId,
  price = 2,
  name = 'product name',
  productQuantity = 2,
  photo = false,
) => {
  const product = createProductItem(
    productId || uuidv4(),
    sellerUsername,
    productQuantity,
    price,
    name,
    photo,
  );

  return {
    _id: uuidv4(),
    quantity,
    product,
  };
};

export const createTransactionAndOrderProdItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
) => ({
  _id: productId || uuidv4(),
  name,
  photo,
  price,
  quantity,
  seller: {
    username: sellerUsername,
  },
});

export const createOrder = (
  products = [createTransactionAndOrderProdItem()],
  orderId = uuidv4(),
  sellerUsername,
  buyerUsername,
  overallPrice,
  createdAt,
  sellerEmail,
  sellerPhone,
) => {
  const seller = sellerUsername
    ? {
        username: sellerUsername,
        email: sellerEmail,
        phone: sellerPhone,
      }
    : null;

  const buyer = buyerUsername
    ? {
        username: buyerUsername,
      }
    : null;

  return {
    _id: orderId,
    seller,
    buyer,
    products,
    overallPrice,
    deliveryAddress: defaultDeliveryAddress,
    createdAt,
    updatedAt: createdAt,
    __v: 0,
  };
};

export const testStore = (initialAuth = {}, initialProduct = {}, initialUI = {}) => {
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
  const defaultState = createStoreWithMiddleware(rootReducer).getState();

  const finalState = {
    auth: {
      ...defaultState.auth,
      ...initialAuth,
    },
    product: {
      ...defaultState.product,
      ...initialProduct,
    },
    ui: {
      ...defaultState.ui,
      ...initialUI,
    },
  };
  const finalStore = createStoreWithMiddleware(rootReducer, finalState);

  return {
    store: finalStore,
    initialState: finalStore.getState(),
  };
};

export const checkReqMethodAndURL = (moxiosInstance, method, url, reqNumber = 0) => {
  if (reqNumber) {
    const requests = moxiosInstance.requests.__items;
    const prevRequest = requests[reqNumber - 1];
    return method === prevRequest.config.method && url === prevRequest.config.url;
  }
  const lastRequest = moxiosInstance.requests.mostRecent();
  return method === lastRequest.config.method && url === lastRequest.config.url;
};

export const createExpectedState = (
  initialState,
  expectedAuth = {},
  expectedProduct = {},
  expectedUI = {},
) => {
  const newState = {
    auth: {
      ...initialState.auth,
      ...expectedAuth,
    },
    product: {
      ...initialState.product,
      ...expectedProduct,
    },
    ui: {
      ...initialState.ui,
      ...expectedUI,
    },
  };

  return newState;
};

export const setUpStoreWithDefaultProfile = (auth = {}, product = {}, ui = {}) => {
  return testStore(
    {
      profile: defaultUserProfile,
      deliveryAddress: defaultDeliveryAddress,
      cart: [],
      ...auth,
    },
    product,
    ui,
  );
};
