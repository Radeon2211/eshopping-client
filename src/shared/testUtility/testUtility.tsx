/* eslint-disable import/no-extraneous-dependencies */
import { ReactElement } from 'react';
import { Router } from 'react-router-dom';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { MockStoreEnhanced } from 'redux-mock-store';
import { fireEvent, waitFor, render } from '@testing-library/react';
import { v4 as uuidv4 } from 'uuid';
import moxios from 'moxios';
import rootReducer from '../../store/reducers/rootReducer';
import theme from '../../styled/theme';
import {
  CartItem,
  DeliveryAddress,
  Order,
  OrderProductItem,
  Product,
  Profile,
  ProfileStatus,
} from '../types/types';
import { ProductCondition } from '../types/enums';

export const defaultDeliveryAddress: DeliveryAddress = {
  firstName: 'firstName',
  lastName: 'lastName',
  street: 'street',
  zipCode: 'zipCode',
  city: 'city',
  country: 'country',
  phone: 'phone',
};

export const defaultUserProfile: Profile = {
  ...defaultDeliveryAddress,
  _id: 'userId',
  username: 'username',
  email: 'email@domain.com',
  status: ProfileStatus.ACTIVE,
  contacts: {
    email: true,
    phone: true,
  },
  createdAt: '2021-02-10T19:10:38.872Z',
};

export const createPaginationProps = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  quantityPerPage: 2,
});

export const createProductItem = (props: Partial<Product> = {}): Product => {
  return {
    _id: uuidv4(),
    quantity: 1,
    price: 2,
    name: 'product name',
    photo: false,
    quantitySold: 0,
    buyerQuantity: 0,
    description: '',
    condition: ProductCondition.NEW,
    seller: {
      username: 'username',
    },
    createdAt: '2021-02-10T19:10:38.872Z',
    updatedAt: '2021-02-10T19:10:38.872Z',
    __v: 0,
    ...props,
  };
};

export const createCartItem = (props = {}): CartItem => {
  const finalProps = {
    id: uuidv4(),
    sellerUsername: 'username',
    quantity: 1,
    productId: uuidv4(),
    price: 2,
    name: 'product name',
    productQuantity: 2,
    photo: false,
    ...props,
  };
  const product = createProductItem({
    _id: finalProps.productId,
    quantity: finalProps.productQuantity,
    price: finalProps.price,
    name: finalProps.name,
    photo: finalProps.photo,
    seller: {
      username: finalProps.sellerUsername,
    },
  });

  return {
    _id: finalProps.id,
    quantity: finalProps.quantity,
    product,
  };
};

export const createTransactionAndOrderProdItem = (
  props: Partial<OrderProductItem> = {},
): OrderProductItem => {
  return {
    _id: uuidv4(),
    quantity: 1,
    price: 2,
    name: 'product name',
    photo: false,
    seller: {
      username: 'username',
    },
    ...props,
  };
};

export const createOrder = (props: Partial<Order> = {}): Order => {
  return {
    _id: uuidv4(),
    overallPrice: 100,
    seller: null,
    buyer: null,
    products: [createTransactionAndOrderProdItem()],
    deliveryAddress: defaultDeliveryAddress,
    createdAt: '2021-02-10T19:10:38.872Z',
    updatedAt: '2021-02-10T19:10:38.872Z',
    __v: 0,
    ...props,
  };
};

export const testStore = (initialAuth = {}, initialProduct = {}, initialUI = {}) => {
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
  const defaultState = createStoreWithMiddleware(rootReducer).getState();

  const finalState = {
    auth: {
      ...defaultState.auth,
      ...initialAuth,
    },
    product: {
      ...(defaultState.product as object),
      ...initialProduct,
    },
    ui: {
      ...(defaultState.ui as object),
      ...initialUI,
    },
  };
  // @ts-expect-error TBF after redux upgrade
  const finalStore = createStoreWithMiddleware(rootReducer, finalState);

  return {
    store: finalStore,
    initialState: finalStore.getState(),
  };
};

type MoxiosRequestsItems = { config: { method: string; url: string } }[];
type MockedMoxiosInstance = {
  requests: {
    __items?: MoxiosRequestsItems;
    mostRecent?: () => { config: { method: string; url: string } };
  };
};
type DefaultMoxiosInstance = Omit<typeof moxios, 'requests'> & {
  requests: typeof moxios.requests & {
    __items?: MoxiosRequestsItems;
  };
};
export const checkReqMethodAndURL = (
  moxiosInstance: MockedMoxiosInstance | DefaultMoxiosInstance,
  method: string,
  url: string,
  reqNumber = 0,
) => {
  if (reqNumber) {
    const requests = moxiosInstance.requests.__items;
    const prevRequest = requests?.[reqNumber - 1];
    return prevRequest && method === prevRequest.config.method && url === prevRequest.config.url;
  }
  if (!moxiosInstance.requests.mostRecent) return false;
  const lastRequest = moxiosInstance.requests.mostRecent();
  return method === lastRequest.config.method && url === lastRequest.config.url;
};

export const createExpectedState = (
  initialState: {
    auth?: object;
    product?: object;
    ui?: object;
  },
  expectedAuth = {},
  expectedProduct = {},
  expectedUI = {},
) => {
  const newState = {
    auth: {
      ...initialState.auth,
      ...expectedAuth,
    },
    product: {
      ...initialState.product,
      ...expectedProduct,
    },
    ui: {
      ...initialState.ui,
      ...expectedUI,
    },
  };

  return newState;
};

export const setUpStoreWithDefaultProfile = (auth = {}, product = {}, ui = {}) => {
  return testStore(
    {
      profile: defaultUserProfile,
      deliveryAddress: defaultDeliveryAddress,
      cart: [],
      ...auth,
    },
    product,
    ui,
  );
};

export const clickAtSubmitButton = async (container: Element) => {
  await waitFor(() => {
    const submitBtn = container.querySelector('button[type="submit"]');
    if (submitBtn) {
      fireEvent.click(submitBtn);
    }
  });
};

export const testRouterPushCall = (
  pushFn: ReturnType<typeof jest.fn>,
  callIdx: number,
  expectedPathname: string,
  expectedSearch = '',
) => {
  expect(pushFn.mock.calls[callIdx][0].pathname).toEqual(expectedPathname);
  expect(pushFn.mock.calls[callIdx][0].search).toEqual(expectedSearch);
};

type RenderAppPartOptions = {
  push?: ReturnType<typeof jest.fn>;
  pathname?: string;
  search?: string;
  withoutThemeProvider?: boolean;
  withoutRouter?: boolean;
  store?: MockStoreEnhanced<object, object>;
  onlyReturnWrappedElement?: boolean;
};

export const renderAppPart = (element: ReactElement, options: RenderAppPartOptions = {}) => {
  const navigator = {
    createHref: jest.fn(),
    push: options.push || jest.fn(),
    go: jest.fn(),
    replace: jest.fn(),
  };

  const location = {
    pathname: options.pathname || '/',
    search: options.search || '',
  };

  let content = element;
  if (!options.withoutThemeProvider) {
    content = <ThemeProvider theme={theme}>{element}</ThemeProvider>;
  }
  if (options.store) {
    content = <Provider store={options.store}>{content}</Provider>;
  }
  if (!options.withoutRouter) {
    content = (
      <Router location={location} navigator={navigator}>
        {content}
      </Router>
    );
  }

  return options.onlyReturnWrappedElement ? content : render(content);
};
