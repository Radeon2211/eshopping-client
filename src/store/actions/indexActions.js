export {
  registerUser,
  loginUser,
  fetchProfile,
  logoutUser,
  fetchOtherUser,
  setOtherUser,
  updateUser,
  changeDeliveryAddress,
  deleteAccount,
  addAdmin,
  removeAdmin,
  sendAccountVerificationLink,
  resetPassword,
} from './authActions';

export {
  fetchCart,
  addCartItem,
  updateCartItem,
  clearCart,
  removeCartItem,
  setTransaction,
  goToTransaction,
  buyProducts,
} from './tradeActions';

export { setModal, deleteMessage, changeProductsPerPage } from './uiActions';

export {
  addProduct,
  editProduct,
  fetchProducts,
  fetchProductDetails,
  setProductDetails,
  deleteProduct,
} from './productActions';

export { fetchOrders, fetchOrderDetails, setOrderDetails } from './orderActions';
