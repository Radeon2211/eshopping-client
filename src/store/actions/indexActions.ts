export {
  registerUser,
  loginUser,
  fetchProfile,
  logoutUser,
  fetchOtherUser,
  setOtherUser,
  updateUser,
  changeEmail,
  changeDeliveryAddress,
  deleteAccount,
  addAdmin,
  removeAdmin,
  sendAccountVerificationLink,
  resetPassword,
} from './authActions/authActions';

export {
  fetchCart,
  addCartItem,
  updateCartItem,
  clearCart,
  removeCartItem,
  setTransaction,
  goToTransaction,
  buyProducts,
} from './tradeActions/tradeActions';

export { setMessage, setModal, changeProductsPerPage } from './uiActions/uiActions';

export {
  addProduct,
  editProduct,
  fetchProducts,
  fetchProductDetails,
  setProductDetails,
  deleteProduct,
} from './productActions/productActions';

export { fetchOrders, fetchOrderDetails, setOrderDetails } from './orderActions/orderActions';
