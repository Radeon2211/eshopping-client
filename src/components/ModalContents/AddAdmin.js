import React, { useCallback } from 'react';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { useDispatch } from 'react-redux';
import * as actions from '../../store/actions/indexActions';
import Form from '../UI/Form/Form';
import Input from '../UI/Input/Input';
import { inputKinds, userRules } from '../../shared/constants';

const validationSchema = Yup.object({
  email: userRules.email,
});

const AddAdmin = () => {
  const dispatch = useDispatch();
  const onAddAdmin = useCallback((email) => dispatch(actions.addAdmin(email)), [dispatch]);

  return (
    <Formik
      initialValues={{
        email: '',
      }}
      validationSchema={validationSchema}
      onSubmit={(data) => {
        onAddAdmin(data.email);
      }}
    >
      {({ dirty, errors, touched, isValid, setFieldTouched }) => (
        <Form btnText="Add" headingText="Add admin" isValid={dirty && isValid} cancellable>
          <Input
            kind={inputKinds.INPUT}
            config={{
              type: 'email',
              name: 'email',
              id: 'email',
              placeholder: 'Email of user that you want make an admin',
              autoComplete: 'email',
              onInput: setFieldTouched.bind(this, 'email', true, true),
            }}
            isValid={!errors.email}
            isTouched={touched.email}
            label="Email"
          />
        </Form>
      )}
    </Formik>
  );
};

export default AddAdmin;
