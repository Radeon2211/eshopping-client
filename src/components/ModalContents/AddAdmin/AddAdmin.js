import { useCallback } from 'react';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { useDispatch } from 'react-redux';
import * as actions from '../../../store/actions/indexActions';
import Form from '../../UI/Form/Form';
import Input from '../../UI/Input/Input';
import { userRules } from '../../../shared/constants';
import { InputKind } from '../../../shared/types/types';

const validationSchema = Yup.object({
  email: userRules.email,
});

export default function AddAdmin() {
  const dispatch = useDispatch();
  const onAddAdmin = useCallback((email) => dispatch(actions.addAdmin(email)), [dispatch]);

  return (
    <Formik
      initialValues={{
        email: '',
      }}
      validationSchema={validationSchema}
      onSubmit={(data) => {
        onAddAdmin(data.email);
      }}
    >
      {({ dirty, errors, touched, isValid, setFieldTouched }) => (
        <Form btnText="Add" headingText="Add admin" isValid={dirty && isValid} cancellable>
          <Input
            kind={InputKind.INPUT}
            config={{
              type: 'email',
              name: 'email',
              id: 'email',
              placeholder: 'Email of user that you want make an admin',
              autoComplete: 'email',
              autoFocus: true,
              onInput: setFieldTouched.bind(this, 'email', true, true),
              'data-testid': 'AddAdmin-email',
            }}
            isValid={!errors.email}
            isTouched={touched.email}
            label="Email"
          />
        </Form>
      )}
    </Formik>
  );
}
