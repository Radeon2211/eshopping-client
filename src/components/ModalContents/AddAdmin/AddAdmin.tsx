import { useCallback } from 'react';
import { Formik } from 'formik';
import * as Yup from 'yup';
import * as actions from '../../../store/actions/indexActions';
import Form from '../../UI/Form/Form';
import Input from '../../UI/Input/Input';
import { userRules } from '../../../shared/constants';
import { InputKind } from '../../../shared/types/types';
import { useTypedDispatch } from '../../../store/reducers/rootReducer';

const validationSchema = Yup.object({
  email: userRules.email,
});

export default function AddAdmin() {
  const dispatch = useTypedDispatch();
  const onAddAdmin = useCallback((email: string) => dispatch(actions.addAdmin(email)), [dispatch]);

  return (
    <Formik
      initialValues={{
        email: '',
      }}
      validationSchema={validationSchema}
      onSubmit={(data) => {
        onAddAdmin(data.email);
      }}
    >
      {({ dirty, errors, touched, isValid, setFieldTouched }) => (
        <Form btnText="Add" headingText="Add admin" isValid={dirty && isValid} cancellable>
          <Input
            kind={InputKind.INPUT}
            config={{
              type: 'email',
              name: 'email',
              id: 'email',
              placeholder: 'Email of user that you want make an admin',
              autoComplete: 'email',
              autoFocus: true,
              onInput: setFieldTouched.bind(AddAdmin, 'email', true, true),
              'data-testid': 'AddAdmin-email',
            }}
            isValid={!errors.email}
            isTouched={touched.email}
            label="Email"
          />
        </Form>
      )}
    </Formik>
  );
}
