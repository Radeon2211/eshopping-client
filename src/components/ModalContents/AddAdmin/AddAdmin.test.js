import React from 'react';
import { render, cleanup, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import AddAdmin from './AddAdmin';
import theme from '../../../styled/theme';
import * as actions from '../../../store/actions/indexActions';
import { clickAtSubmitButton } from '../../../shared/testUtility/testUtility';

const mockStore = configureMockStore([thunk]);

const setUp = () => {
  const store = mockStore({
    ui: {
      isFormLoading: false,
      formError: '',
    },
  });
  store.dispatch = jest.fn();

  return {
    ...render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <AddAdmin />
        </ThemeProvider>
      </Provider>,
    ),
    store,
  };
};

jest.mock('../../../store/actions/indexActions.js', () => ({
  ...jest.requireActual('../../../store/actions/indexActions.js'),
  addAdmin: (email) => email,
}));

afterEach(cleanup);

describe('<AddAdmin />', () => {
  describe('check form', () => {
    it('should have empty input and focus on it by default', () => {
      setUp();
      const emailInput = screen.getByTestId('AddAdmin-email');
      expect(emailInput).toHaveFocus();
      expect(emailInput.value).toEqual('');
    });

    it('should call addAdmin() with given email after input submit and button click', async () => {
      const { store, container } = setUp();

      const emailInput = screen.getByTestId('AddAdmin-email');
      const testEmail = 'test@domain.com';

      await waitFor(() => {
        fireEvent.change(emailInput, { target: { value: testEmail } });
      });
      expect(emailInput.value).toEqual(testEmail);

      await clickAtSubmitButton(container);
      expect(store.dispatch).toHaveBeenNthCalledWith(1, actions.addAdmin(testEmail));

      await waitFor(() => {
        fireEvent.submit(emailInput);
      });
      expect(store.dispatch).toHaveBeenNthCalledWith(2, actions.addAdmin(testEmail));
    });

    it('should NOT call resetPassword() if input is empty', async () => {
      const { store, container } = setUp();
      await clickAtSubmitButton(container);
      expect(store.dispatch).not.toHaveBeenCalled();
    });

    it('should NOT call resetPassword() if email is not valid', async () => {
      const { store, container } = setUp();

      const emailInput = screen.getByTestId('AddAdmin-email');
      const testEmail = 'invalidemail';

      await waitFor(() => {
        fireEvent.change(emailInput, { target: { value: testEmail } });
      });
      expect(emailInput.value).toEqual(testEmail);

      await clickAtSubmitButton(container);

      expect(store.dispatch).not.toHaveBeenCalled();
    });
  });
});
