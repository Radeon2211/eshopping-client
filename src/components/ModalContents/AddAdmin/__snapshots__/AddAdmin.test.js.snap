// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddAdmin /> check how renders should render everything correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hYjfqi"
  >
    <h3
      class="sc-iBPRYJ jtekve"
    >
      Add admin
    </h3>
    <form
      action="#"
      class="formik-form"
    >
      <div
        class="sc-fubCfw drZTsR"
        type="email"
      >
        <label
          class="sc-jrAGrp bkXExJ"
          for="email"
        >
          Email
        </label>
        <input
          autocomplete="email"
          class="sc-pFZIQ fjonmB"
          data-testid="AddAdmin-email"
          id="email"
          name="email"
          placeholder="Email of user that you want make an admin"
          type="email"
          value=""
        />
      </div>
      <div
        class="sc-gKsewC dLOeOj"
        direction="row"
        spacing="3"
        width="100%"
        wrap="nowrap"
      >
        <span
          class="sc-eCssSg gABWUT"
          color=""
          display="block"
          overflow="visible"
          spacing="0px"
          transform=""
        >
          <button
            class="sc-gsTCUz gNMffe"
            color="blue"
            type="button"
          >
            cancel
          </button>
        </span>
        <button
          class="sc-gsTCUz ghohWv"
          color="blue"
          data-testid="Form-submit-btn"
          disabled=""
          type="submit"
        >
          Add
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
